src/py/flwr/client/node_state.py

```diff
@@ -48,4 +61,9 @@ def retrieve_context(self, run_id: int) -> Context:
 
     def update_context(self, run_id: int, context: Context) -> None:
         """Update run context."""
+        if context.run_config != self._initial_run_config:
```

Comment:
This needs to do a deep comparison with the initial `run_config` for this `run_id`